### DISCUSSION TITLE
Integration screen could use condensing & optimizing

### DISCUSSION THREAD

**Discussion by vvuk** [home-assistant/frontend#8636] [Configuration]
When I first started using HA a long while ago, I had very few integrations.  As HA & connected homes have exploded, the number of integrations is quite large but the tile-based view can't really keep up. Sample screenshot here, though I know everyone's familiar with this screen:
![image](https://user-images.githubusercontent.com/194965/111058757-652d2780-8445-11eb-854c-5a2911d86e67.png)

There are a issues:
- Integrations that have very few devices (sometimes only one) are given the same visual "weight" as integrations with many devices.  That screenshot is from the top of my list, and I don't interact with any of those integrations at all (okay maybe the HomeKit one).
- Each integration tile is very large, but for when you do need to interact with an integration, it's usually through the relatively small device/entity links, that are a very small part of the overall
- "Rename" is an action that's pretty rarely taken -- it could go in the three-dots menu per integration.  (Same with "Options" really, but see next point)
- There's an "Options" on the tile, sometimes a "Configure", and then a "System Options" in the three-dots menu that do different things, and the difference on what they mean or what to find where is confusing to a user.

The logos are pretty and it's great to see everything that's connected, but they're not super useful over time.  A few suggestions & ideas:

- Add ability to sort this display by number of devices/entities
- Add some way to deprioritize/lower integrations, which would put them at the end, maybe even in a separately collapsed area.  For example, I would put `weather`, `hacs`, the `monoprice` amp, and a whole bunch of things there -- I don't need to interact with the integration itself, I can just interact with the devices that they've created.
- Maybe as a simpler version to all of the above, just add a way to "pin" some integrations at the top of the list, separate from unpinned ones?
- Move Rename to three-dots menu
- Sort out & consolidate "Options"/"Configure" and "System Options", and move to three-dots menu
- Always have every Integration have an "Integration" page, just like devices have a devices page.  At a bare minimum it can just list all the devices/entities on it.
- Change to more list-based or other denser layout, increasing hit targets for more common areas

**Comment by SeanPM5** [comment]
I don't think sorting by device/entities is all that useful for two main reasons:

1. The number of entities an integration has doesn't always equate to importance. As an example the "Xbox" integration adds 8 sensors for every person on your friends list (so 10 friends would mean 80 entities). Integrations like this would always be listed first, even if less significant to the user.
2. These numbers fluctuate, meaning your Integrations page could potentially be ordered differently every time you visit it. That can be unpredictable and messy, and if the user isn't able to form any muscle memory on where things are placed, it will also require more reading and thus be slower to find things.

Alphabetical sort is better IMO as it's natural and requires zero explanation. There's always the search bar if you don't feel like scrolling. Even as an optional sort, it'd require extra UI and maintenance for something that I think people wouldn't use (for the reasons listed above).

I agree that "Rename" isn't frequently used, but the tradeoff to putting both that and "Options" button in the three-dot / overflow menu is that it'll be less discoverable there. And in this quick mockup below you can see that moving Rename and Options to that menu doesn't really save any space or result in a more condensed view.

![options](https://i.imgur.com/cHmMcEC.png)

**Reply by vvuk** [reply]
True, sorting by # of entities is not really that great.  But some kind of sort where you can actually get the "important" integrations more easily accessible is useful; that's where the pin etc. suggestion comes from.

For Rename/Options -- it doesn't save space, but it does reduce cognitive load when looking at this screen, regardless of the style.  Fewer highlighted things.  Also, the biggest hit target is the icon/graphic and name for each integration -- it should really go somewhere.  I'd suggest a list of devices, or if none/just one, then the list of entities.  Even better would be an integration screen like the device screen that shows all of the things provided by that integration, and can incorporate options etc. all in one screen; but that's a bigger change.


(And actually consolidating the different Options properly would just reduce confusion)
